### PR TITLE
Fix generationTargetQ freezing and invalidating for LfBus

### PR DESCRIPTION
**Pull request by SylvestreSakti**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] A PR or issue has been opened in all impacted repositories (if any)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**

#1240 introduced a new formalism for bus targetQ updating. This new formalism was not correctly applied for secondary voltage control, triggering an exception when restoring a bus state that has been modified with secondary voltage control outer loop.
`com.powsybl.commons.PowsyblException: Generation targetQ cannot be frozen if generatorVoltageControl is enabled`

This fix led to a refactoring to avoid such errors

**What is the current behavior?**
The exception is thrown

**What is the new behavior (if this is a feature change)?**
When activating voltage control, targetQ is correctly invalidated by the secondary voltage control outer loop, there is no exception.

Moreover, to avoid this kind of mistake, the methods `invalidateGenerationTargetQ()` and `setGenerationVoltageControlEnabled(bool)` are now private. Outside the class `AbstractLfBus`, we can use : 
- `setGeneratorVoltageControlEnabledAndRecomputeTargetQ(bool) `: activates or deactivates generationVoltageControl and invalidates targetQ (if targetQ was frozen, it turns unfrozen)
- `freezeGenerationTargetQAndDisableGeneratorVoltageControl(double) `: deactivates generatorVoltageControl and freezes targetQ (used to switch a bus from PV to PQ)

Two test cases are introduced : 
1 - The exception throwing case due to Secondary Voltage Control Outer loop
2 - An uncovered corner case : Security Analysis with a contingency on a generator that switched PV -> PQ in the base case

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java [hunk 1]
```diff
@@ -282,6 +282,11 @@ public void setGeneratorVoltageControlEnabled(boolean generatorVoltageControlEna
     @Override
     public void setVoltageControlEnabled(boolean enabled) {
         setGeneratorVoltageControlEnabled(enabled);
```
Comment (vidaldid-rte) [inline src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java]: You may also nos make setGeneratorVoltageControl private, and remove it from the interface,  as calling it alone may craete inconsistencies

[the remark this answers is not in the document]
Comment (SylvestreSakti) [reply src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java]: Finally we left it as it is, using this LfBus method `setVoltageControlEnabled` only for the VoltageTargetChecker which that not depend interfere with generatorTargetQ
